Source/BoundaryConditions/PEC_Insulator.H

```diff
@@ -0,0 +1,174 @@
+
+#include <AMReX_Array.H>
+#include <AMReX_Geometry.H>
+#include <AMReX_Vector.H>
+
+#include <AMReX_BaseFwd.H>
+
+#include <array>
+#include <memory>
+
+class PEC_Insulator
+{
+public:
+
+    PEC_Insulator();
+
+    /**
+     * \brief Apply either the PEC or insulator boundary condition on the boundary and in the
+     *        guard cells.
+     *        In the PEC, the nodal fields (in a Yee mesh) as made even relative to the boundary,
+     *        the non-nodal fields are made odd.
+     *        In the insulator, the tangential fields are set to the value if specified, otherwise unchanged,
+     *        and the normal fields extrapolated from the valid cells.
+     *
+     * \param[in,out] Efield
+     * \param[in]     ng_fieldgather      number of guard cells used by field gather
+     * \param[in]     geom                geometry object of level "lev"
+     * \param[in]     lev                 level of the Multifab
+     * \param[in]     patch_type          coarse or fine
+     * \param[in]     ref_ratios          vector containing the refinement ratios of the refinement levels
+     * \param[in]     time                current time of the simulation
+     * \param[in]     split_pml_field     whether pml the multifab is the regular Efield or
+     *                                    split pml field
+     */
+    void ApplyPEC_InsulatortoEfield (std::array<amrex::MultiFab*, 3> Efield,
+                                     amrex::Array<FieldBoundaryType,AMREX_SPACEDIM> const & field_boundary_lo,
+                                     amrex::Array<FieldBoundaryType,AMREX_SPACEDIM> const & field_boundary_hi,
+                                     amrex::IntVect const & ng_fieldgather, amrex::Geometry const & geom,
+                                     int lev, PatchType patch_type, amrex::Vector<amrex::IntVect> const & ref_ratios,
+                                     amrex::Real time,
+                                     bool split_pml_field = false);
+    /**
+     * \brief Apply either the PEC or insulator boundary condition on the boundary and in the
+     *        guard cells.
+     *        In the PEC, the nodal fields (in a Yee mesh) as made even relative to the boundary,
+     *        the non-nodal fields are made odd.
+     *        In the insulator, the tangential fields are set to the value if specified, otherwise unchanged,
+     *        and the normal fields extrapolated from the valid cells.
+     *
+     * \param[in,out] Bfield
+     * \param[in]     ng_fieldgather      number of guard cells used by field gather
+     * \param[in]     geom                geometry object of level "lev"
+     * \param[in]     lev                 level of the Multifab
+     * \param[in]     patch_type          coarse or fine
+     * \param[in]     ref_ratios          vector containing the refinement ratios of the refinement levels
+     * \param[in]     time                current time of the simulation
```

Comment:
Could you add the docstrings for `field_boundary_lo` and `field_boundary_hi` as well?